chore: change blockquotes

harish-sethuraman opened the pull request:
Followup to https://github.com/babel/website/pull/2730
searched:
```
<blockquote class="alert alert--info">((.|\n)*?)<\/blockquote>
```
replaced with 
```
:::info$1:::
```

searched:
```
<blockquote class="alert alert--warning">((.|\n)*?)<\/blockquote>
```
replaced with 
```
:::caution$1:::
```

searched:
```
<blockquote class="alert alert--danger">((.|\n)*?)<\/blockquote>
```
replaced with 
```
:::danger$1:::
```

JLHwung commented on website/data/tools/babel_node_debug/usage.md:
````diff
@@ -8,9 +8,9 @@ Or use the alias:
 ```
 
-<blockquote class="alert alert--info">
+:::info
````
Can you check if this is working? The `/data/**/*.md` are rendered by react-markdown, which may not enable the remark plugin required by admonitions `:::`.

harish-sethuraman replied:
<img width="1322" alt="Screenshot 2023-02-25 at 12 54 02 AM" src="https://user-images.githubusercontent.com/32865581/221272558-43bd459b-42d5-44ed-8f55-27b2f8134fa4.png">
Interestingly it seems to work 😅

JLHwung replied:
Can you check https://deploy-preview-2731--babel.netlify.app/setup#installation and click jspm?

These files do not support hot loading because they are not declared as external webpack dependencies.

harish-sethuraman replied:
Then is it fine to replace only inside the docs and blog directory?

JLHwung commented on website/data/tools/babel_register/usage.md:
```diff
@@ -14,22 +14,22 @@ import "@babel/register";
 
-<blockquote class="alert alert--warning">
+:::caution
   <h4>Not suitable for libraries</h4>
```
Can you use move heading to the admonition title? e.g.

```md
:::caution Not suitable for libraries
...
:::
```

harish-sethuraman replied:
Just wanted to confirm if we are going to do this for everything? We have a mix of both h3 and h4 headings inside the admonition.

JLHwung replied:
Yes I think generally the heading should be the title if we migrate to admonition.

Comparing https://deploy-preview-2731--babel.netlify.app/docs/learn with https://babel.dev/docs/learn.

This PR:
<img width="979" alt="image" src="https://user-images.githubusercontent.com/3607926/221287416-85fce302-2b5f-441f-afdb-5bd8e6b78f59.png">

Current main:
<img width="1009" alt="image" src="https://user-images.githubusercontent.com/3607926/221287577-c9d119fc-edcf-4b63-8015-d1598350e678.png">

The admonition will take up more spaces when both the admonition type and the heading are displayed.

About DocsSearch: currently we configured the agolia bot to crawl `main h3` and `main h4` as lv3 and lv4 indices. I will update the crawler config after this PR gets merged.

JLHwung commented on docs/learn.md:
```diff
@@ -3,20 +3,20 @@ title: Learn ES2015
 ---
 
-<blockquote class="alert alert--info">
+:::info
```
Prettier currently do not support admonitions: https://github.com/prettier/prettier/issues/8640 Fortunately we are not affected yet because we are not using the `proseWrap` option.

harish-sethuraman commented:
I have done the following changes:
- changed the block quotes to the admonitions syntax. 
- changed headings to admonition syntax heading
- remove `<p>`
- remove tabs as it converted the pargraphs to code blocks

JLHwung commented on docs/learn.md:
```diff
@@ -479,21 +461,15 @@ System.get("jquery");
-</blockquote>
+:::info Using Module Loader
+In order to use this, you'll need to tell Babel to use the
+<code>system</code> module formatter. Also be sure to check out <a href="https://github.com/systemjs/systemjs">System.js</a>.
```
Can you remove the line break before `<code>`? Docusaurus will generate a `</p>` here.

JLHwung commented on docs/learn.md:
```diff
@@ -715,13 +673,10 @@ Two new numeric literal forms are added for binary (`b`) and octal (`o`).
-</blockquote>
+:::caution Only supports literal form
+Babel is only able to transform <code>0o767</code> and not
+<code>Number("0o767")</code>.
```
ditto.

harish-sethuraman replied:
Im just wondering why these were in new line? May be we should set-up some linting rules for all these?

JLHwung replied:
I have no idea. I guess it has to be dealt with how MDX parses the source.

> May be we should set-up some linting rules for all these?

Good idea, I am wondering are there any such rules out there now.

harish-sethuraman replied:
https://github.com/mdx-js/eslint-mdx - This is what we are looking for?

JLHwung commented on docs/learn.md:
```diff
@@ -697,15 +658,12 @@ Array.of(1, 2, 3) // Similar to new Array(...), but without special one-arg beha
+:::caution Limited support from polyfill
+Most of these APIs are supported by the Babel <a href="/docs/babel-polyfill">polyfill</a>. However, certain
+features are omitted for various reasons (e.g.
+<code>String.prototype.normalize</code> needs a lot of additional code to
```
ditto.

JLHwung reviewed (approved):
Thanks!
